### PR TITLE
Add RSS Feed Generation Script

### PR DESCRIPTION
**Pull request by yan-pi**
**Description:**
This PR introduces a script to generate an RSS feed for the blog. The script reads the markdown files from the `src/articles` directory, extracts metadata using `gray-matter`, and creates an RSS feed using the `rss` library. The feed is then written to the `public/rss.xml` file.

**Changes Made:**
- Added `dotenv` to handle environment variables.
- Created `gen-rss.js` script to generate the RSS feed.
- Updated the `package.json` with a new script command: `generate:rss`.
- Configured the script to use `SITE_URL` from environment variables for generating URLs.

**How to Test:**
1. Ensure you have a `.env.local` file in the project root with the following content:
   ```
   SITE_URL=https://lukeberrypi.vercel.app/
   ```
2. Run the following command to generate the RSS feed:
   ```bash
   npm run generate:rss
   ```
3. Verify that `public/rss.xml` is generated and contains the correct URLs and post data.

**Notes:**
- The script is designed to be run locally and in production, using environment variables to configure the site URL.
- In development, it defaults to `http://localhost:3000` if `SITE_URL` is not set.

### CODE REVIEW

File: src/articles/frontend-debugging-101.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 title: "Frontend Debugging 101"
-date: "23-07-2024"
+date: "2024-07-23"
```
Comment (LukeberryPi) [inline src/articles/frontend-debugging-101.md:3]: this format change appears to have broken the dates in the /articles/page.tsx

could you update /lib/articles.ts to the new format?

File: .env.example [hunk 2]
```diff
@@ -0,0 +1 @@
+SITE_URL=https://lukeberrypi.vercel.app/
```
Comment (LukeberryPi) [inline .env.example]: might be good to use VERCEL_PROJECT_PRODUCTION_URL from here https://vercel.com/docs/projects/environment-variables/system-environment-variables#system-environment-variables

**Review by LukeberryPi — CHANGES_REQUESTED** [review]
- env nit
- article dates broken

**Comment by yan-pi** [comment]
Of course!! I will also correct the gen-rss script. I noticed some issues with reading in certain RSS readers, but I truly liked your review. I will apply the improvements.

File: src/articles/frontend-debugging-101.md (hunk 1 above)
Comment (LukeberryPi) [reply src/articles/frontend-debugging-101.md:3]: keep it as yyyy-mm-dd, i prefer it like that 👍

**Review by LukeberryPi — APPROVED** [review]
lgtm!

**Pull request merged**